### ISSUE TITLE
Hardhat v2.0.9 throws `Callback was already called.` errors when testing

### ISSUE THREAD

**Issue by nlordell** [nomiclabs/hardhat#1237]
I recently tried bumping from Hardhat v2.0.8 to v2.0.9 and noticed all my tests started failing with:
```
     Uncaught Error: Callback was already called.
```

At first I thought it might have been some issue with my project setup, but it seems to also affect the template project:

```
$ mkdir hardhat-test; cd hardhat-test
$ yarn init -y
$ yarn add -D hardhat
$ npx hardhat init # Create sample project
...
$ npx hardhat test
...
     Uncaught Error: Callback was already called.
      at /home/foobar/Developer/hardhat-test/node_modules/merkle-patricia-tree/node_modules/async/lib/async.js:43:36
      at WriteStream.<anonymous> (node_modules/merkle-patricia-tree/node_modules/async/lib/async.js:358:17)
      at WriteStream.destroy (node_modules/level-ws/level-ws.js:140:8)
      at finish (_stream_writable.js:670:14)
      at processTicksAndRejections (internal/process/task_queues.js:80:21)
...
```

Some environment information (feel free to ask if I missed something important):
```
$ npx hardhat --version
2.0.9
$ node --version
v14.15.1
$ yarn --version
1.22.10
$ cat /etc/redhat-release 
Fedora release 33 (Thirty Three)
```

**Comment by anudit** [comment]
Same here, was looking forward to `eth_signTypedData_v4` support in hardhat.

**Comment by fvictorio** [comment]
Thanks @nlordell. Definitely a bug. As a temporary workaround, this doesn't seem to happen with node 12.

**Comment by bitmammoth** [comment]
Seems related to this:

https://github.com/trufflesuite/ganache-core/pull/636/commits/99cf1ee7251a8e7cc0421a662503a8bc0a24e3a0

https://github.com/trufflesuite/ganache-core/issues/568

**Comment by alcuadrado** [comment]
This was a regression that we accidentally introduced in v2.0.9. We are working on fixing it at the moment.

**Comment by fvictorio** [comment]
This should be fixed on 2.0.10.

**Comment by nlordell** [comment]
Works like a charm!

Just wanted to say kudos for the awesome tool! I really enjoy using Hardhat 😄
